src/server/frontend/wayland/wayland_connector.cpp

```diff
@@ -610,24 +615,26 @@ class WlSurface : public wayland::Surface
             session->destroy_buffer_stream(stream_id);
     }
 
-    void set_resize_handler(std::function<void(geom::Size)> const& handler)
+    void set_resize_handler(std::function<void(Size)> const& handler)
     {
         resize_handler = handler;
     }
 
-    void set_hide_handler(std::function<void()> const& handler)
+    void set_hide_handler(std::function<void(bool visible)> const& handler)
```

Comment:
So this will result in hide_handler(true) actually making it visible? Confusing!

Reply:
I'll rename it